packages/purpose-process/src/services/purposeService.ts

```diff
@@ -545,6 +566,66 @@ export function purposeServiceBuilder(
         totalCount: purposesList.totalCount,
       };
     },
+    async createPurpose(
+      purposeSeed: ApiPurposeSeed,
+      organizationId: TenantId,
+      correlationId: string
+    ): Promise<{ purpose: Purpose; isRiskAnalysisValid: boolean }> {
+      logger.info(
+        `Creating Purpose for EService ${purposeSeed.eserviceId} and Consumer ${purposeSeed.consumerId}`
+      );
+      const eserviceId = unsafeBrandId<EServiceId>(purposeSeed.eserviceId);
+      const consumerId = unsafeBrandId<TenantId>(purposeSeed.consumerId);
+      assertOrganizationIsAConsumer(organizationId, consumerId);
+
+      assertConsistentFreeOfCharge(
+        purposeSeed.isFreeOfCharge,
+        purposeSeed.freeOfChargeReason
+      );
+
+      const tenant = await retrieveTenant(organizationId, readModelService);
+
+      assertTenantKindExists(tenant);
+
+      const validatedFormSeed = validateAndTransformRiskAnalysis(
+        purposeSeed.riskAnalysisForm,
+        tenant.kind
+      );
+
+      await retrieveActiveAgreement(eserviceId, consumerId, readModelService);
+
+      const purposeWithSameName = await readModelService.getSpecificPurpose(
+        eserviceId,
+        consumerId,
+        purposeSeed.title
+      );
+
+      if (purposeWithSameName) {
+        throw duplicatedPurposeTitle(purposeSeed.title);
+      }
+
+      const purpose: Purpose = {
+        ...purposeSeed,
+        id: generateId(),
+        createdAt: new Date(),
+        eserviceId,
+        consumerId,
+        versions: [
+          {
+            id: generateId(),
+            state: purposeVersionState.draft,
+            dailyCalls: purposeSeed.dailyCalls,
+            createdAt: new Date(),
+          },
+        ],
+        riskAnalysisForm: validatedFormSeed,
+      };
+
+      await repository.createEvent(
+        toCreateEventPurposeAdded(purpose, correlationId)
+      );
+      return { purpose, isRiskAnalysisValid: validatedFormSeed !== undefined };
```

Comment:
@AsterITA should `isRiskAnalysisValid` be `true`? Otherwise it is raised as an error during the validation, isn't it?

Reply:
Or instead should we use a function that doesn't throw the error?

Reply:
Right, it can be set to `true`!